conans/test/functional/profiles/test_python_profiles.py

```diff
@@ -0,0 +1,164 @@
+import os
+import textwrap
+import unittest
+from collections import defaultdict
+
+from importlib.util import spec_from_file_location, module_from_spec
+
+from conans.client.tools import environment_append
+from conans.test.utils.tools import TestClient
+from conans import ProfileSettings, ProfileOptions
+
+
+class ProfileLoader(object):
+
+    @staticmethod
+    def _init_profile():
+        import conans
+        conans.settings = ProfileSettings()
+        conans.options = ProfileOptions()
+        conans.env = ProfileOptions()
+
+    @staticmethod
+    def _load_module_from_file(filename):
+        name = os.path.splitext(os.path.basename(filename))[0]
+        spec = spec_from_file_location(name, filename)
+        config = module_from_spec(spec)
+        spec.loader.exec_module(config)
+        return config
+
+    @staticmethod
+    def _load_profile(profile_path):
+        profile = {}
+        profile_content = ProfileLoader._load_module_from_file(profile_path)
+        for key, value in profile_content.__dict__.items():
+            if not key.startswith("__"):
+                profile[key] = value
+        return profile.copy()
+
+    @staticmethod
+    def load_profile_python(profile_path, parent_profile_path=None):
+        # TODO: implement include() of other profiles? variable substitution?
+        if parent_profile_path:
+            ProfileLoader._load_profile(parent_profile_path)
+        profile = ProfileLoader._load_profile(profile_path)
+        ProfileLoader._init_profile()
+        return profile.copy()
+
+
+class PythonProfilesTest(unittest.TestCase):
+
+    def setUp(self):
+        self.client = TestClient()
+        self.pr1_content = textwrap.dedent("""
+            import os
+            from conans import settings, options, env
```

Comment:
This import of a global object, and the way the object is re-initialized for each profile doesn't look good.

It would be cleaner to do something like:

```python
from conan.profiles import Settings, Options  # To be define where it is, definitely in the new ``conan.`` scope, but not in the root
from conan.tools.env import Environment # the new one

settings = Settings()
settings.os = "Windows"

env = Environment()
env.append_path("PATH", os.getenv("MyLocalPathVar"))

```

Reply:
I agree, the re-initilialization is not very elegant that way but wanted to try the clean syntax of not having to initialize a new object in the profile file. 

I think it is easier to just have `seettings` initialized and you can apply the values without having to "compose" the "Settings()" object. You only load the files in the given order and just apply pure python rules.

Reply:
Not worth if we incur in such code anti-pattern, this access to global objects is bad, and we have also had very bad experience with it.